packages/rainbowkit/src/components/ConnectButton/abbreviateETHBalance.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Adapted from https://github.com/domharrington/js-number-abbreviate
+ */
+const units = ['k', 'm', 'b', 't'];
+
+export function abbreviateETHBalance(number: number): string {
+  if (number < 1) return number.toFixed(3).replace(/0+$/, '');
+
+  if (number < 10 ** 4)
+    return new Intl.NumberFormat()
+      .format(parseFloat(number.toFixed(1)))
+      .replace(/\.0$/, '');
+
+  const decimalsDivisor = 10 ** 1; // 1 decimal place
+
+  let result = String(number);
+
+  for (let i = units.length - 1; i >= 0; i--) {
+    const size = 10 ** ((i + 1) * 3);
+
+    if (size <= number) {
+      number = Math.round((number * decimalsDivisor) / size) / decimalsDivisor;
+
+      // round up
```

Comment:
We talked about offline (posting here for others to see), but rounding here is my biggest concern. 

- if you have `0.5999`, do you really want to see that you have `0.6` vs `0.599`? 
  - I would say you want to see this "rounded down", because it's misleading and potentially upsetting to see that you don't have as much money as you hoped (expectations not met), vs potentially being surprised that you have a minuscule amount _more_. 
  - I'm trying to "think like a bank" here, since users may be using this in similar style applications.